app/controllers/financialOverview/PsaPaymentsAndChargesController.scala

```diff
@@ -0,0 +1,119 @@
+import uk.gov.hmrc.viewmodels.Text.Message
+import viewmodels.Table
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class PsaPaymentsAndChargesController @Inject()(
+                                                 override val messagesApi: MessagesApi,
+                                                 identify: IdentifierAction,
+                                                 allowAccess: AllowAccessActionProviderForIdentifierRequest,
+                                                 val controllerComponents: MessagesControllerComponents,
+                                                 psaPenaltiesAndChargesService: PsaPenaltiesAndChargesService,
+                                                 financialStatementConnector: FinancialStatementConnector,
+                                                 minimalConnector: MinimalConnector,
+                                                 renderer: Renderer
+                                               )(implicit ec: ExecutionContext)
+  extends FrontendBaseController
+    with I18nSupport
+    with NunjucksSupport {
+
+  private val logger = Logger(classOf[PsaPaymentsAndChargesController])
+
+  def onPageLoad(journeyType: ChargeDetailsFilter): Action[AnyContent] =
+    (identify andThen allowAccess()).async { implicit request =>
+        val response = for {
+          psaName <- minimalConnector.getPsaOrPspName
+          psaFSWithPaymentOnAccount <- financialStatementConnector.getPsaFSWithPaymentOnAccount(request.psaIdOrException.id)
+          penaltiesCache <- psaPenaltiesAndChargesService.getPenaltiesForJourney(request.psaIdOrException.id, journeyType)
+        } yield {
+          val psaFSWithoutPaymentOnAccount: Seq[PsaFSDetail] = psaFSWithPaymentOnAccount.seqPsaFSDetail.filterNot(c => c.chargeType ==
+            PsaFSChargeType.PAYMENT_ON_ACCOUNT || c.chargeType == REPAYMENT_INTEREST)
+          renderFinancialOverdueAndInterestCharges(psaName, request.psaIdOrException.id, psaFSWithoutPaymentOnAccount,
+            request, psaFSWithPaymentOnAccount.seqPsaFSDetail, journeyType, penaltiesCache)
+        }
+        response.flatten
+    }
+
+  //scalastyle:off parameter.number
+  //scalastyle:off method.length
+  //scalastyle:off cyclomatic.complexity
+  private def renderFinancialOverdueAndInterestCharges(psaName: String,
+                                                       psaId: String,
+                                                       psaFS: Seq[PsaFSDetail],
+                                                       request: RequestHeader,
+                                                       creditPsaFS: Seq[PsaFSDetail],
+                                                       journeyType: ChargeDetailsFilter,
+                                                       penaltiesCache: PenaltiesCache)
+                                                      (implicit messages: Messages, headerCarrier: HeaderCarrier): Future[Result] = {
+
```

Comment:
I know you have not introduced this change, would you mind if you create a refactor ticket to change the psaPenaltiesAndChargesService.retrievePsaChargesAmount to return a case class rather than a tuple of three.

Reply:
https://jira.tools.tax.service.gov.uk/browse/PODS-6849